proxy/http/HttpSessionManager.cc

```diff
@@ -398,7 +404,9 @@ HttpSessionManager::_acquire_session(sockaddr const *ip, CryptoHash const &hostn
     // Now check to see if we have a connection in our shared connection pool
     EThread *ethread = this_ethread();
     Ptr<ProxyMutex> pool_mutex =
-      (TS_SERVER_SESSION_SHARING_POOL_THREAD == pool_type) ? ethread->server_session_pool->mutex : m_g_pool->mutex;
+      (TS_SERVER_SESSION_SHARING_POOL_THREAD == sm->t_state.http_config_param->server_session_sharing_pool) ?
```

Comment:
`pool_type` that is passed by the caller is used after acquiring the lock. Doesn't it cause mismatch?

Reply:
Good point. That does seem to be an error.